cpp/src/arrow/util/async_util.cc

```diff
@@ -274,7 +282,9 @@ class AsyncTaskSchedulerImpl : public AsyncTaskScheduler {
       AbortUnlocked(submit_result.status(), std::move(lk));
       return;
     }
-    submit_result->AddCallback([this, cost](const Status& st) {
+    // FIXME(C++17, move into lambda?)
```

Comment:
should be able to do this now?

Reply:
Yep.  Although, in retrospect, I'm not sure how it works, since `std::function` is copyable (maybe it works just because we don't happen to copy it)